app/src/main/java/ar/com/wolox/android/example/ui/example/root/RootActivity.java

```diff
@@ -15,25 +25,70 @@
 
     @Override
     protected void init() {
+        validateLogin();
+    }
+
+    @Override
+    protected int layout() {
+        return R.layout.activity_base;
+    }
+
+    private void validateLogin() {
         SharedPreferences sharedPreferences = this.getSharedPreferences(
                 getString(R.string.shared_preference_login), Context.MODE_PRIVATE);
-        boolean loginStatus = sharedPreferences.getBoolean(getString(
-                R.string.shared_preference_login_status), false);
-
-        Intent intent;
+        String email = sharedPreferences.getString(getString(R.string.shared_preference_email),
+                "");
+        String password = sharedPreferences.getString(getString(
+                R.string.shared_preference_password), "");
+        boolean wasLoggedIn = sharedPreferences.getBoolean(
+                getString(R.string.shared_preference_login_status), false);
 
-        if (loginStatus) {
-            intent = new Intent(this, HomeActivity.class);
+        /* This check is to optimize avoiding the request to the REST API */
+        if (!wasLoggedIn || TextUtils.isEmpty(email)) {
+            launchLoginActivity();
         } else {
-            intent = new Intent(this, LoginActivity.class);
+           validateLoginWithDatabase(email, password);
         }
+    }
 
-        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
-        startActivity(intent);
+    private void validateLoginWithDatabase(String email, String password) {
+        Retrofit retrofit = new Retrofit.Builder()
+                .baseUrl(getString(R.string.rest_api_base_url))
+                .addConverterFactory(GsonConverterFactory.create())
+                .build();
+        UserService userService = retrofit.create(UserService.class);
+        Call<List<User>> userCall = userService.getUserByEmail(email);
+        userCall.enqueue(new NetworkCallback<List<User>>() {
+            @Override
+            public void onResponseSuccessful(@Nullable List<User> users) {
+                Intent intent;
+
+                if (users != null && !users.isEmpty() && users.get(0).getPassword().
+                        equals(password)) {
+                    intent = new Intent(getApplicationContext(), HomeActivity.class);
```

Comment:
Create two functions goToHomeScreen and goToLoginScreen. I don't like the idea of using intents inside request.

Reply:
✔ **Done**

I thoughed the same, I had even done the launchLoginActivity() function, but then I forgot to do the same for HomeActivity :disappointed: